lms/lmsdb/models.py

```diff
@@ -209,6 +237,24 @@ def on_save_handler(model_class, instance, created):
         instance.api_key = generate_password_hash(instance.api_key)
 
 
+class UserCourse(BaseModel):
+    user = ForeignKeyField(User, backref='usercourses')
+    course = ForeignKeyField(Course, backref='usercourses')
+    date = DateTimeField(default=datetime.now)
+
+    @classmethod
+    def is_user_registered(cls, user_id: int, course_id: int) -> bool:
+        return (
+            cls.
+            select()
+            .where(
+                cls.user == user_id,
+                cls.course == course_id,
+            )
+            .exists()
+        )
```

Comment:
I'd implement this on `User` and on `Course` (`User.is_registered` and `Course.has_user`), or maybe even in `Models` if possible